src/btreemap/iter.rs

```diff
@@ -158,10 +153,102 @@ where
                                     next: Index::Entry(idx),
                                 });
+                                break;
+                            } else {
+                                // We found the key that we must
+                                // exclude. We add its left neighbor
+                                // to the stack and start iterating
+                                // from its left child.
+                                let left_child = match node.node_type() {
+                                    NodeType::Internal => Some(node.child(idx)),
+                                    NodeType::Leaf => None,
+                                };
+
+                                if idx > 0 && self.range.contains(node.key(idx - 1)) {
+                                    self.backward_cursors.push(Cursor::Node {
+                                        node,
+                                        next: Index::Entry(idx - 1),
+                                    });
+                                }
+                                if let Some(left_child) = left_child {
+                                    self.backward_cursors.push(Cursor::Address(left_child));
+                                }
+                                break;
+                            }
+                        }
+                        Err(idx) => {
+                            // The `idx` variable points to the first
+                            // key that is greater than the left
+                            // bound.
+                            //
+                            // If the index points to a valid node, we
+                            // will visit its left subtree and then
+                            // return to this key.
+                            //
+                            // If the index points at the end of
+                            // array, we'll continue with the right
+                            // child of the last key.
+
+                            // Load the left child of the node to visit if it exists.
+                            // This is done first to avoid cloning the node.
+                            let child = match node.node_type() {
+                                NodeType::Internal => {
+                                    // Note that loading a child node cannot fail since
+                                    // len(children) = len(entries) + 1
+                                    Some(self.map.load_node(node.child(idx)))
+                                }
+                                NodeType::Leaf => None,
+                            };
+
+                            if idx > 0 && self.range.contains(node.key(idx - 1)) {
+                                self.backward_cursors.push(Cursor::Node {
+                                    node,
+                                    next: Index::Entry(idx - 1),
+                                });
```

Comment:
I'm not entirely sure about the above comments but it feels like we need to adjust something as there's some diff in what we do here vs in the forward cursors case.